src/storage/observer.rs

```diff
@@ -146,6 +145,11 @@ where
             .await
     }
 
+    pub(crate) async fn try_fsync_data(&self) {
+        self.send_msg(Msg::new(OperationType::TryFsyncData, None))
```

Comment:
There is a problem here: after the `dirty_bytes` exceeds the threshold, messages will be sent on each new record, which can lead to queue overflow. An additional message frequency limit is required. 
One possible approach is to store the last time the message was sent and ignore sending new once if some constant amount of time was not passed (e.g. 500ms). Time measurement can be done with `coarsetime` crate. Time can be stored in `AtomicU64` as milliseconds elapsed from start of the epoch (Unix time).
Another approach is to track, that `fsync` is in progress with `AtomicBool` inside `Inner<K>` struct and don't send new messages if it is in progress.

Reply:
I agree, I'll implement the latter aproach